app/Services/SiteService.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\V2\Sites\Site;
+use App\StateMachines\EntityStatusStateMachine;
+use App\StateMachines\SiteStatusStateMachine;
+
+class SiteService
+{
+    public static function updateSiteStatus($site_uuid)
+    {
+        if (! $site_uuid) {
+            return;
+        }
+        $site = Site::where('uuid', $site_uuid)->first();
+        if ($site->status != EntityStatusStateMachine::APPROVED) {
+            return;
+        }
+        $site->status = SiteStatusStateMachine::RESTORATION_IN_PROGRESS;
+        $site->save();
+    }
```

Comment:
This method is doing the work that the transitions and state machine methods should be doing instead. For one thing, the status should never be updated directly. Instead, it should be updated with `$site->status()->transitionTo()` (and then the `save()` call is not needed. 

The pattern established for the other state machines is to make this a method right on the model (or a trait for the model) as well. See for instance, `updateInProgress()` or `approve()` in `HasReportStatus`. Instead of having a service for this, I'd have the controller pull the record from the DB and call a method like `restorationInProgress()` on the `Site` instance, and that method would look something like
```
public function restorationInProgress()
{
  if ($site->status == EntityStatusStateMachine::RESTORATION_IN_PROGRESS) {
    $site->status()->transitionTo(SiteStatusStateMachine::RESTORATION_IN_PROGRESS);
  }
}
```

I don't love the check above for the current state. Ideally this method would only be called when the site _should_ be in the correct state, and then the `transitions` defined on the state machine will throw an error if the transition isn't allowed, but I don't know these flows well enough to know if that will cause problems.

Reply:
make use of `transitionTo`

Reply:
name changed